workorder-job-repeater.lua

```diff
@@ -0,0 +1,296 @@
+-- reduces pauses between jobs of infinite work orders
+
+local function print_help()
+    print[====[
+
+workorder-job-repeater
+======================
+This script recreates workshop jobs upon completion in case
+they come from a work order with amount set to infinite. Usually
+a job for an infinite work order is only created once a day,
+resulting in dwarfs lazying around (this doesn't apply to non-infinite
+work orders). This script makes the dwarfes work harder.
+
+For best experience add following to your ``onMapLoad*.init``::
+
+    workorder-job-repeater start
+    
+**Usage**:
+
+``[enable | disable] workorder-job-repeater``
+
+``workorder-job-repeater start [<frequency>] | stop``
+
+``workorder-job-repeater --frequency <ticks> | --verbose | --help``
+
+:enable, start <frequency>: starts job monitoring, running every ``<frequency>`` ticks (default: 10).
+:disable, stop:             stops the script.
+
+-f, --frequency <ticks>     sets job monitoring frequency to ``<ticks>``
+-v, --verbose               toggles script's verbosity.
+-h, --help                  this help.
+
+]====]
+end
+
+local utils = require 'utils'
+local eventful = require 'plugins.eventful'
+
+local DEBUG = 1
```

Comment:
A Lua debug logging framework is useful beyond just this script. Could this be factored out into a general API that could be used by all scripts and configured with `debugfilter`?

Reply:
I can think of a way to make this into a Lua module/script, but I have no idea how to make it configurable with `debugfilter`. I can give it a try after I finish my other pending projects.

Reply:
Yeah, definitely out of scope for this PR. We already cross the Lua/cpp interface for printing, but I'm not sure how we could efficiently make use of the cpp message filtering. It might end up being best to filter locally in Lua and just sync filter configs between the cpp and Lua implementations. But yeah, that's a discussion for another PR.